=== modified file 'tools/build/v2/tools/gcc.jam'
--- tools/build/v2/tools/gcc.jam	2013-12-29 13:13:00 +0000
+++ tools/build/v2/tools/gcc.jam	2013-12-29 14:36:31 +0000
@@ -635,7 +635,7 @@
 
 actions compile.asm
 {
-    "$(CONFIG_COMMAND)" $(LANG) $(OPTIONS) -D$(DEFINES) -I"$(INCLUDES)" -c -o "$(<)" "$(>)"
+    "$(CONFIG_COMMAND)" $(LANG) $(OPTIONS) -D$(DEFINES)  $(USER_OPTIONS) -I"$(INCLUDES)" -c -o "$(<)" "$(>)"
 }
 
 # Class checking that we do not try to use the <runtime-link>static property

