=== modified file 'libs/context/build/Jamfile.v2'
--- libs/context/build/Jamfile.v2	2013-12-29 11:28:14 +0000
+++ libs/context/build/Jamfile.v2	2013-12-29 11:31:05 +0000
@@ -414,6 +414,25 @@
    ;
 
 alias asm_context_sources
+#   : asm/make_i386_x86_64_sysv_macho_gas.S
+#     asm/jump_i386_x86_64_sysv_macho_gas.S
+#   : <address-model>32_64
+#     <architecture>x86
+#     <binary-format>mach-o
+#     <target-os>darwin
+#     <toolset>darwin
+#   ;
+#
+#alias asm_context_sources
+   : [ make asm/make_i386_x86_64_sysv_macho_gas.o : asm/make_i386_x86_64_sysv_macho_gas.S : @gas ]
+     [ make asm/jump_i386_x86_64_sysv_macho_gas.o : asm/jump_i386_x86_64_sysv_macho_gas.S : @gas ]
+   : <address-model>32_64
+     <architecture>x86
+     <binary-format>mach-o
+     <target-os>darwin
+   ;
+
+alias asm_context_sources
    : asm/make_x86_64_ms_pe_masm.asm
      asm/jump_x86_64_ms_pe_masm.asm
      dummy.cpp

=== added file 'libs/context/src/asm/jump_i386_x86_64_sysv_macho_gas.S'
--- libs/context/src/asm/jump_i386_x86_64_sysv_macho_gas.S	1970-01-01 00:00:00 +0000
+++ libs/context/src/asm/jump_i386_x86_64_sysv_macho_gas.S	2013-12-29 11:33:46 +0000
@@ -0,0 +1,16 @@
+/*
+            Copyright Sergue E. Leontiev 2013.
+   Distributed under the Boost Software License, Version 1.0.
+      (See accompanying file LICENSE_1_0.txt or copy at
+          http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+// Stub file for universal binary
+
+#if defined(__i386__)
+    #include "jump_i386_sysv_macho_gas.S"
+#elif defined(__x86_64__)
+    #include "jump_x86_64_sysv_macho_gas.S"
+#else
+    #error "No arch's"
+#endif

=== added file 'libs/context/src/asm/make_i386_x86_64_sysv_macho_gas.S'
--- libs/context/src/asm/make_i386_x86_64_sysv_macho_gas.S	1970-01-01 00:00:00 +0000
+++ libs/context/src/asm/make_i386_x86_64_sysv_macho_gas.S	2013-12-29 11:32:49 +0000
@@ -0,0 +1,16 @@
+/*
+            Copyright Sergue E. Leontiev 2013.
+   Distributed under the Boost Software License, Version 1.0.
+      (See accompanying file LICENSE_1_0.txt or copy at
+          http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+// Stub file for universal binary
+
+#if defined(__i386__)
+    #include "make_i386_sysv_macho_gas.S"
+#elif defined(__x86_64__)
+    #include "make_x86_64_sysv_macho_gas.S"
+#else
+    #error "No arch's"
+#endif

