--- src/osx/carbon/dataobj.cpp	2016-09-17 10:26:10.000000000 +1200
+++ src/osx/carbon/dataobj.cpp	2016-09-17 10:26:23.000000000 +1200
@@ -458,13 +458,11 @@ bool wxDataObject::GetFromPasteboard( vo
                                 pastelocationset = true;
                            }
                         }
-#if 0 // See https://groups.google.com/forum/#!topic/wx-dev/wFxevpvbhvQ/discussion
                         else if ( flavorFormat.GetType() != wxDF_PRIVATE )
                         {
                             // indicate the expected format for the type, benefiting from native conversions eg utf8 -> utf16
                             flavorType = (CFStringRef) wxDataFormat( flavorFormat.GetType()).GetFormatId();
                         }
-#endif
 
                         err = PasteboardCopyItemFlavorData( pasteboard, itemID, flavorType , &flavorData );
                         if ( err == noErr )
