=== modified file 'src/common/translation.cpp'
--- src/common/translation.cpp	2014-01-26 13:59:40 +0000
+++ src/common/translation.cpp	2014-01-26 15:18:47 +0000
@@ -1611,6 +1611,9 @@
 /* static */
 const wxString& wxTranslations::GetUntranslatedString(const wxString& str)
 {
+    // Marco Serantoni - Dirty Patch
+    // Awaiting for http://trac.wxwidgets.org/ticket/15908
+    if( gs_translationsOwned == false) return str; 
     wxLocaleUntranslatedStrings& strings = wxThreadInfo.untranslatedStrings;
 
     wxLocaleUntranslatedStrings::iterator i = strings.find(str);

